src/core/auth/client/AuthClient.ts

```diff
@@ -86,31 +79,36 @@ export class AuthClient {
   ): Promise<true | string | undefined> {
     const response = await this._apiClient.login(user, password);
     if (response && typeof response === 'object' && 'token' in response) {
-      await this.processAuthResponse(response);
       // reload to init with authenticated config and profile
       location.reload();
       return true;
     }
     return response;
   }
 
-  public async checkAuth(): Promise<boolean> {
+  public checkAuth() {
+    const response = this.checkLocalAuthToken();
+    if (response?.token) {
+      // auth init flow
+      this.startAuthenticated(response);
+      return true;
+    }
+    // anon init flow
+    this.startPublic();
+    return false;
+  }
+
+  startPublic() {
+    over(this.loginHooks)();
+  }
```

Comment:
i like anonimus word here very much!
may be method should be called `startAnonymosly` too?

Reply:
why not)